tools/SetupFlow/DevHome.SetupFlow/ViewModels/QuickstartPlaygroundViewModel.cs

```diff
@@ -430,73 +446,102 @@ public void UpdateProgress(QuickStartProjectProgress progress)
         }
     }
 
+    [RelayCommand]
+    public async Task SubmitChat()
+    {
+        var message = CustomPrompt;
+
+        if (message != null && message != string.Empty)
+        {
+            ShowExamplePrompts = false;
+            ChatMessages.Add(new ChatStyleMessage
+            {
+                Name = message,
+                Type = ChatStyleMessage.ChatMessageItemType.Request,
+            });
+
+            await GenerateChatStyleCompetions(message);
+
+            CanGenerateCodespace = true;
+        }
+    }
+
     [RelayCommand(CanExecute = nameof(CanGenerateCodespace))]
     public async Task GenerateCodespace()
     {
-        try
+        if (!string.IsNullOrEmpty(CustomPrompt) && ActiveQuickstartSelection != null)
         {
-            var userPrompt = CustomPrompt;
+            try
+            {
+                // Clear chat
+                ChatMessages.Clear();
 
-            // TODO: Replace this (and throughout) with proper diagnostics / logging
-            ArgumentNullException.ThrowIfNullOrEmpty(userPrompt);
-            ArgumentNullException.ThrowIfNull(ActiveQuickstartSelection);
+                var userPrompt = CustomPrompt;
 
-            TelemetryFactory.Get<ITelemetry>().Log("QuickstartPlaygroundGenerateButtonClicked", LogLevel.Critical, new GenerateButtonClicked(userPrompt), ActivityId);
+                // TODO: Replace this (and throughout) with proper diagnostics / logging
+                ArgumentNullException.ThrowIfNullOrEmpty(userPrompt);
+                ArgumentNullException.ThrowIfNull(ActiveQuickstartSelection);

```

Comment:
How can `ActivateQuickStartSelection` be null if the only way this code runs is when `ActivateQuickStartSelection` is not null?